typescript/commands/references.py

```diff
@@ -196,23 +194,45 @@ def run(self, text, argsJson):
         self.view.settings().set('refinfo', ref_info.as_value())
         self.view.set_read_only(True)
 
-
-def get_lines_from_file(fileName, entries):
-    line_nos = defaultdict(list)
-    for entry in entries:
-        line_nos[entry["start"]["line"] - 1].append(entry)
-    f = None
-    try:
-        f = open(fileName, encoding='utf-8', errors='ignore')
-        count = 0
-        for line in f:
-            if line_nos.get(count):
-                for entry in line_nos.get(count):
-                    entry["lineText"] = line.rstrip()
-            count = count + 1
-
-    except Exception as e:
-        log.exception('Error fetching preview from %s' % (fileName))
-    finally:
-        if f is not None:
-            f.close()
+class RefsHelper:
```

Comment:
I remember we talked about this one before - either approach seems fine, I just worry about the performance implications of this approach.

__lines and __content will be maps that contain every file and every file -> line mapping for _all_ references simultaneously, which seems like it could blow up ram.

That being said, it's easier to debug and reason about, so can understand that approach.

Have you tried on an example with 100 references?